appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java

```diff
@@ -7,48 +7,96 @@
-      e.printStackTrace();
       return false;
     }
   }
 
-  public static void promptToInstallWallet(Activity activity, Context context, String message,
-      DialogVisibleListener dialogVisibleListener) {
-    showWalletInstallDialog(activity,context,message,dialogVisibleListener);
-  }
+  public static void promptToInstallWallet() {
+    final Activity act;
+    try {
+      act = getActivity();
+    } catch (Exception e) {
+      e.printStackTrace();
+      return;
+    }
 
-  private static void showWalletInstallDialog( final Activity activity,Context context,String message,final DialogVisibleListener dialogVisibleListener) {
+    if (act == null) {
+      Log.d(WalletUtils.class.getName(),"Problem retrieving main Activity");
+      return;
+    }
 
+    dialogVisible = true;
     AlertDialog.Builder builder;
-    builder = new AlertDialog.Builder(context);
+    builder = new AlertDialog.Builder(act);
     builder.setTitle(R.string.wallet_missing);
-    builder.setMessage(message);
+    builder.setMessage(act.getString(R.string.install_wallet_from_iab));
+    Log.d("String name: ", act.getString(R.string.install_wallet_from_iab));
+
     builder.setPositiveButton(R.string.install, new DialogInterface.OnClickListener() {
-      @Override public void onClick(DialogInterface dialogInterface, int i) {
-        activity.startActivity(
+      @Override public void onClick(DialogInterface dialog, int which) {
+        act.startActivity(
             new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName)));
       }
     });
+
     builder.setNegativeButton(R.string.skip, new DialogInterface.OnClickListener() {
       @Override public void onClick(DialogInterface dialogInterface, int i) {
-        dialogVisibleListener.OnDialogVisibleListener(false);
+        dialogInterface.cancel();
+        dialogVisible = false;
       }
     });
+
     builder.setIcon(android.R.drawable.ic_dialog_alert);
     builder.show();
   }
+
+  public static Activity getActivity() throws Exception {
+    Class activityThreadClass = Class.forName("android.app.ActivityThread");
+    Object activityThread = activityThreadClass.getMethod("currentActivityThread")
+        .invoke(null);
+    Field activitiesField = activityThreadClass.getDeclaredField("mActivities");
+    activitiesField.setAccessible(true);
+    ArrayMap activities = (ArrayMap) activitiesField.get(activityThread);
+    for (Object activityRecord : activities.values()) {
+      Class activityRecordClass = activityRecord.getClass();
+      Field pausedField = activityRecordClass.getDeclaredField("paused");
+      pausedField.setAccessible(true);
+      if (!pausedField.getBoolean(activityRecord)) {
+        Field activityField = activityRecordClass.getDeclaredField("activity");
+        activityField.setAccessible(true);
+        Activity activity = (Activity) activityField.get(activityRecord);
+        return activity;
+      }
+    }
+    return null;
```

Comment:
Since we are already throwing an exception, can we use that to throw an exception instead of null, since the action for both returns (null or exception) is the same.